src/algebra/algebra/subalgebra.lean

```diff
@@ -606,22 +606,24 @@ lemma subsingleton_of_subsingleton [subsingleton A] : subsingleton (subalgebra R
 ⟨λ B C, ext (λ x, by { simp only [subsingleton.elim x 0, zero_mem] })⟩
 
 -- TODO[gh-6025]: make this an instance once safe to do so
```

Comment:
@Vierkantor: how about:
```suggestion
/--
For performance reasons this is not an instance. If you need this instance, add
```
local attribute [instance] alg_hom.subsingleton subalgebra.subsingleton_of_subsingleton
```
in the section that needs it
-/
-- TODO[gh-6025]: make this an instance once safe to do so
```

Reply:
Yeah, actually thinking about it a bit more, it's a much better experience to enable these instances when you need them, rather than writing `haveI` in a tactic block. So we don't really need the shortcut.